kyo-core/shared/src/main/scala/kyo/StreamCoreExtensions.scala

```diff
@@ -61,6 +62,36 @@ object StreamCoreExtensions:
                             _ <- emitMaybeChunksFromChannel(channel)
                         yield ()
 
+        /** Creates a stream from an iterator.
+          *
+          * @param v
+          *   Iterator to create a stream from
+          * @param bufferSize
+          *   Size of the buffer that the iterator will write to and the stream will read from
+          */
+        def fromIterator[V, S](v: => Iterator[V], bufferSize: Int = Stream.DefaultChunkSize)(using
+            tag: Tag[Emit[Chunk[V]]],
+            frame: Frame
+        ): Stream[V, IO] =
+            Stream:
+                IO:
+                    val it      = v
+                    val size    = bufferSize max 1
+                    val builder = ChunkBuilder.init[V]
+
+                    Stream.repeatPresent(
+                        IO.ensure(builder.clear()):
```

Comment:
There's no need to call `clear`. The underlying buffer is released and cleared automatically by `result`.